jcl/src/java.base/share/classes/java/lang/invoke/MethodHandles.java

```diff
@@ -5599,6 +5600,66 @@ static MethodHandle collectReturnValue(MethodHandle target, MethodHandle filter)
 		}
 		return buildTransformHandle(new CollectReturnHelper(target, filter), resultType);
 	}
+	
+	/**
+	 * Return an array of checked exceptions for a MethodHandle.
+	 * 
+	 * @param mh A MethodHandle to scan for checked exceptions.
+	 * 
+	 * @return an array of checked exception classes if any or null.
+	 * 
+	 * @throws InternalError for an error.
+	 */
+	static Class<?>[] checkedExceptions(MethodHandle mh) {
```

Comment:
Also, I think this method should be renamed as currently implemented, it's about exceptions in general, not just checked ones.

Reply:
Renamed to `hasCheckedException` since a boolean is returned.